src/main/resources/com/google/api/codegen/ruby/test.snip

```diff
@@ -122,10 +126,95 @@
   end
 @end
 
+@private serverStreamingTest(test)
+  it 'invokes {@test.clientMethodName} without error' do
+    @if initCode(test.initCode)
+      @# Create request parameters
+      {@initCode(test.initCode)}
+    @end
+    @# Create expected grpc response
+    {@initCode(test.mockResponse.initCode)}
+
+    @# Mock Grpc layer
+    {@mockServerStreamingRequest(test)}
+    mock_stub = MockGrpcClientStub.new(:{@test.clientMethodName}, mock_method)
+
+    {@test.grpcStubCallString}.stub(:new, mock_stub) do
+      client = {@test.serviceConstructorName}.new
+
+      @# Call method
+      {@methodCallWithResponse(test)}
+
+      @# Verify the response
+      assert_equal(1, response.first(2).size) # The enumerable should only have one element.
+      assert_equal(expected_response, response.first)
+    end
+  end
+@end
+
+@private clientStreamingTest(test)
+  it 'invokes {@test.clientMethodName} without error' do
+    @# Create request parameters
+    {@initCode(test.initCode)}
+    requests = [request]
+    @if test.hasReturnValue
+    
+      @# Create expected grpc response
+      {@initCode(test.mockResponse.initCode)}
+    @end
+
+    @# Mock Grpc layer
+    {@mockClientStreamingRequest(test)}
+    mock_stub = MockGrpcClientStub.new(:{@test.clientMethodName}, mock_method)
+
+    {@test.grpcStubCallString}.stub(:new, mock_stub) do
+      client = {@test.serviceConstructorName}.new
+
+      @# Call method
+      {@methodCallWithResponse(test)}
+
+      @# Verify the response
+      @if test.hasReturnValue
+        assert_equal(expected_response, response)
+      @else
+        assert_nil(response)
+      @end
+    end
+  end
+@end
+
+@private bidiStreamingTest(test)
+  it 'invokes {@test.clientMethodName} without error' do
+    @# Create request parameters
+    {@initCode(test.initCode)}
+    requests = [request]
+    @if test.hasReturnValue
+
+      @# Create expected grpc response
+      {@initCode(test.mockResponse.initCode)}
+    @end
+
+    @# Mock Grpc layer
+    {@mockBidiStreamingRequest(test)}
+    mock_stub = MockGrpcClientStub.new(:{@test.clientMethodName}, mock_method)
+
+    {@test.grpcStubCallString}.stub(:new, mock_stub) do
+      client = {@test.serviceConstructorName}.new
+
+      @# Call method
+      {@methodCallWithResponse(test)}
+
+      @# Verify the response
+      assert_equal(1, response.first(2).size) # The enumerable should only have one element.
```

Comment:
Ah blah, read through the enumerable API expecting to see a size or a length method. When I didn't find it I assumed that that kind of functionality didn't exist since enumerables _can_ be of infinite length.